File: tests/benchmarks/cases/call_chain_polymorphic/c/benchmark_case.c
#include "benchmark_case.h"
#include "benchmark_support.h"

static ZrBenchInt zr_bench_case_call_chain_polymorphic_run(int scale) {
    return zr_bench_run_call_chain_polymorphic(scale);
}

const ZrBenchCaseDescriptor zr_bench_case_descriptor_call_chain_polymorphic = {
        "call_chain_polymorphic",
        "BENCH_CALL_CHAIN_POLYMORPHIC_PASS",
        zr_bench_case_call_chain_polymorphic_run
};

File: tests/benchmarks/cases/call_chain_polymorphic/java/benchmark_case.java
final class CallChainPolymorphicCase {
    static final String NAME = "call_chain_polymorphic";
    static final String PASS_BANNER = "BENCH_CALL_CHAIN_POLYMORPHIC_PASS";

    private CallChainPolymorphicCase() {}

    static long run(int scale) {
        return BenchmarkSupport.callChainPolymorphic(scale);
    }
}
